verifier/src/tcb.rs

```diff
@@ -31,16 +31,36 @@
 
 extern crate alloc;
 
+use alloc::boxed::Box;
 use alloc::string::String;
 use alloc::vec::Vec;
+use der::DateTime;
+use p256::ecdsa::signature::Verifier;
+use p256::ecdsa::{Signature, VerifyingKey};
 use serde::Deserialize;
 use serde_json::value::RawValue;
 
 /// Error parsing TCB(Trusted Computing Base) info
 #[derive(displaydoc::Display, Debug)]
 pub enum Error {
+    /// Error converting from DER
+    Der(der::Error),
     /// Error parsing TCB(Trusted Computing Base) json info: {0}
     Serde(serde_json::Error),
+    /// Error decoding the signature in the TCB data
+    SignatureDecodeError,
```

Comment:
Up for suggestions here as `serde_json::Error` doesn't derive `PartialEq` which means this error isn't and it's a bit ergonomic in tests, which means it may be  clunky for clients

Reply:
yeah what i've often done is given up on errors have `PartialEq` because libraries don't consistently implement it, and i tend to use `assert_matches!` or similar when I want to test this stuff.

I think actual users of this API will use a match statement if they want to case out on the errors, so it's not a big deal for them if `PartialEq` is missing I think

Reply:
updated tests to use `assert!(matches!(`